internal/errors/error.go

```diff
@@ -48,15 +48,63 @@ func New(c Code, opt ...Option) error {
 	}
 }
 
+// E builds an Err value from its args.
+// The type of each argument determines its meaning.
+// If more than is one parameter for a type is provided, only the last one is used.
+//
+// The types are:
+//	errors.Code
+//		The class of the error
+//	errors.Op
+//		The operation raising the error, usually formatted as `package.funcName`
+//	string
+//		Error message
+//	error
+//		The wrapped error that triggered this one
+//
+// If Code is not defined, we set it to the wrapped error's Code
+func E(args ...interface{}) error {
+	e := &Err{}
+	for _, arg := range args {
+		switch arg := arg.(type) {
+		case Code:
+			e.Code = arg
+		case Op:
+			e.Op = arg
+		case string:
+			e.Msg = arg
+		case *Err: // order is important, this match must before "case error:"
+			c := *arg
+			e.Wrapped = &c
+		case error:
+			e.Wrapped = arg
+		default:
+			// ignore it
```

Comment:
We should probably log this somewhere, I am not sure how our logging story works any pointers would be great @mgaffney